calico-enterprise/operations/monitor/metrics/recommended-metrics.mdx

```diff
@@ -0,0 +1,514 @@
+---
+
+# Recommended Prometheus metrics to monitor
+
+## Big picture
+
+Monitor the {{prodname}} Typha, Felix, and policy component metrics to ensure optimal cluster operation.
+
+## Concepts
+
+{{prodname}} Typha, Felix, and policy components are the most critical to monitor because they are responsible for ensuring networking and security functions are up-to-date and working as expected.
+
+### Typha
+
+The primary function of Typha is to allow for increased cluster scale by reducing the load on Kubernetes API Server. Without Typha, large clusters (200+ nodes) would need a considerable amount of memory to correspond to the continuous watches and requests from calico-nodes running in the cluster.
+
+Typha is a caching data store proxy that sits between calico-nodes and Kubernetes API Server. It maintains a single data store connection on behalf of all of its clients (processes running in the calico-node pods, with Felix being Typhas’ main client). Typha watches for node, pod, network policy, bgp configuration, and other events on the Kubernetes API Server, caches and deduplicates this data, and fans out these events to its clients.
+
+### Felix
+
+Felix is a component of calico-node and is responsible for {{prodname}} network policy.
+Felix must be continuously in sync with the data store to ensure the correct set of policies are applied to the node it is running on.
+
+![Typha-felix](/img/calico-enterprise/typha-felix.png)
+
+### About metrics
+
+The following section provides threshold values, symptoms indicating a threshold breach, and recommended actions in case of breached thresholds, to maintain optimal cluster operation.
+
+Threshold values for each metric depend on the cluster size and churn rate. Threshold recommendations are provided where possible, but because each cluster is different, and metrics can depend on cluster churn rate and scale, we recommend that you baseline the cluster to establish numbers that represent normal figures for your cluster.
+
+:::note
+
+Investigate metrics that start increasing rapidly from the baseline set.
+
+:::
+
+## Metrics
+
+**Policy**
+- [Policy metrics](#policy-metrics)
+
+**Typha**
+- [Typha general metrics](#typha-general-metrics)
+- [Typha cluster mesh metrics](#typha-cluster-mesh-metrics)
+- [Typha client metrics](#typha-client-metrics)
+- [Typha cache internals](#typha-cache-internals)
+- [Typha snapshot details](#typha-snapshot-details)
+
+**Felix**
+- [Felix cluster state metrics](#felix-cluster-state-metrics)
+- [Felix error metrics](#felix-error-metrics)
+- [Felix time-based metrics](#felix-time-based-metrics)
+
+## Policy metrics
+
+### Denied traffic
+
+| Denied traffic                   |                                                              |
+| -------------------------------- | ------------------------------------------------------------ |
+| Metric                           | *calico_denied_packets / calico_denied_bytes*                |
```

Comment:
I checked it he code and this isn't just one metric, it's actually a family of metrics so you'll get one instance of calico_denied_packets/bytes for each policy rule that's denying traffic:
```
calico_denied_packets{policy="tier1|fv/policy-1|0|deny|-1",srcIP="10.245.13.133"}
```
Presenting it as one entry makes it sound simple but it's quite a tricky one!